src/models/event-timeline-set.js

```diff
@@ -759,7 +759,7 @@ EventTimelineSet.prototype.aggregateRelations = function(event) {
         }
     }
 
-    relationsWithEventType.addEvent(event);
+    relationsWithEventType.addEvent(event, relatesToEvent);
```

Comment:
We should add a separate `setTargetEvent` or similar to do this separately, and also ensure we only do it one time per relations collection (once the target event is found), as `findEventById` is not very efficient.